### DISCUSSION TITLE
The Client is not listening for custom events but debug shows that the event has been received

### DISCUSSION THREAD

**Discussion by draconidss** [socketio/socket.io#4742] [Q&A]
## Versions

- Client: "socket.io-client": "^2.5.0"（js）
- Server: 4.6.1 (python)

## Problem
I use `DEBUG=engine,socket.io* node file.js` to run, the console prints a custom event, but the program doesn't listen to it `multi_terminal_login` (if I'm not describing this in detail, please give me some advice on troubleshooting socketio issues). thanks!

``` js
const io = require('socket.io-client');
var sid_socket = new Map();

function run_socketio_client(uid, client_id) {
    var api = "https//localhost:3000";
    var socketio_client = io(api, {
        transports: ['websocket'],
    });
    socketio_client.on('connect', function () {
        console.log(' --- --- --- connect --- --- ---');
    });
    socketio_client.on('disconnect', function () {
        console.log(' --- --- --- connect --- --- ---');
    });
    socketio_client.on("multi_terminal_login", function (data) {
        console.log(' --- --- --- multi_terminal_login --- --- ---');
    });
}

function single_socket() {
    run_socketio_client();
}

single_socket();
```

<img width="1925" alt="image" src="https://github.com/socketio/socket.io/assets/52524117/2a159bca-93bf-4c12-943d-0298026971dd">

**Comment by darrachequesne** [comment]
Hi! I was not able to reproduce the issue. Could you please provide a complete example based on https://github.com/socketio/socket.io-fiddle?

From your screenshot, the packet looks well formatted though.
